### PR TITLE
Initial version of the Unity Editor

### PR DESCRIPTION
**Pull request by jurf**
**Important note:** I am not willing to maintain this, I simply do not have the resources (bandwidth, storage space), time or use case.

It uses extra-data because AFAIK according to the EULA (was like that a year ago, didn’t check if still) Unity can’t be hosted elsewhere.

So far it runs, but there are still many things to be done, but seen as quite a lot of people use Unity on Linux (at least my [past](https://github.com/Unity3D-Wine-Support/Unity3D-on-Wine) [experiences](https://aur.archlinux.org/packages/unity-editor/) suggest that), so I think it’ll get the ball rolling.

### To-do list
- [x] Basic UnityEditor manifest
- [x] AppData
- [ ] Test on a real project (e.g. editing code)
- [ ] **Extensions:** 
  - [ ] Documentation
  - [ ] Standard Assets
  - [ ] Example Project
  - [ ] Android Build Support
  - [ ] iOS Build Support
  - [ ] Mac Build Support
  - [ ] WebGL Build Support
  - [ ] Windows Build Support
  - [ ] Facebook Gameroom Build Support

New releases are announced [here](https://forum.unity.com/threads/unity-on-linux-release-notes-and-known-issues.350256/page-2). Their installer uses an `.ini` file containing all the information needed, newest one [here](https://beta.unity3d.com/download/aea5ecb8f9fd+/unity-2017.3.1f1-linux.ini).

Documentation on Flatpak extensions is scarce though and I haven’t really figured out yet how they work, I might add several extensions once I do.

### CODE REVIEW

**Comment by nedrichards** [comment]
I'm happy to help maintain this.

**Comment by nedrichards** [comment]
bot, build com.unity.UnityEditor

**Comment by nedrichards** [comment]
bot, build com.unity.UnityEditor

**Comment by nedrichards** [comment]
Pushed icons and appdata. Seems to launch and log into my account but I haven't had a real play with it yet.

**Comment by nedrichards** [comment]
bot, build com.unity.UnityEditor

**Comment by nedrichards** [comment]
bot, build com.unity.UnityEditor

**Comment by jurf** [comment]
Hey, thanks for that!

**Comment by jurf** [comment]
bot, build com.unity.UnityEditor

**Comment by nedrichards** [comment]
I may have been quicker, but I don't think mine actually worked - looking at the log. Which is strange to say the least.

**Comment by jurf** [comment]
Have you been able to run any project containing scripts? I just get an empty error message in the console and a popup saying compiler errors need to be fixed before running when I try doing so.

I looked at the release topic in the forum and found this list of dependencies:
* `gtk2` (`libgtk2.0-0`)
* `libsoup` (`libsoup2.4-1`)
* `libarchive` (`libarchive13`)
* `libpng` (`libpng12` or later)
* `libgconf` (`libgconf-2-4`)

So I believe we’re still missing `libarchive` and `libpng`? Maybe that’s the culprit?

**Comment by nedrichards** [comment]
Yeah, I got that popup on initial testing but haven't had time to go back and look as to why. I was going to see if any of the normal 'sdk' steps needed to be taken for a flatpak (e.g. --devel etc.)

**Comment by jurf** [comment]
https://forum.unity.com/threads/empty-error-in-console.509173/

Looks like we need to bundle Mono…

**Comment by jurf** [comment]
I tried adding the mono5 extension but no luck…

**Comment by nedrichards** [comment]
Oh, do we need Monodevelop inside the sandbox? Calling @directhex to the mono aisle...

**Comment by jurf** [comment]
I don’t get it though, they’ve just deprecated MonoDevelop.

**Comment by dylanmtaylor** [comment]
You're shipping a really old version of Unity here - https://beta.unity3d.com/download/77a142ac9989/public_download.html is the latest Linux public beta mentioned on https://forum.unity.com/threads/unity-on-linux-release-notes-and-known-issues.350256/page-2

**Comment by dylanmtaylor** [comment]
Oh, I see you want the .tar.xz file ---- http://beta.unity3d.com/download/77a142ac9989/LinuxEditorInstaller/Unity.tar.xz

**Comment by dylanmtaylor** [comment]
I made a pull request against @jurf's branch to pull in the latest version. I was able to build and run the app.

**Comment by jurf** [comment]
Thanks!

BTW, were you able to actually run a project? I still have no idea what could be wrong there.

**Comment by nedrichards** [comment]
bot, build com.unity.UnityEditor

**Comment by jurf** [comment]
@nedrichards it’s only an extra-data change, will the rebuild even show anything?

**Comment by nedrichards** [comment]
So, I've tested this and I can build and run projects in the latest version. Their new IDE strategy for Linux appears to be VS Code or JetBrains Rider: https://blogs.unity3d.com/2018/01/05/discontinuing-support-for-monodevelop-unity-starting-in-unity-2018-1/

This is unfortunate as Flatpak really isn't very good at sharing things between different flatpaks. I'm going to try hacking VS Code into the flatpak and seeing if I can get something working.

As an aside @dylanmtaylor is this project something you're interested in working on and helping to improve? Patches welcome!

File: com.unity.UnityEditor.json [hunk 1]
```diff
@@ -53,11 +53,9 @@
                 {
                     "type": "extra-data",
                     "filename": "Unity.tar.xz",
-                    "md5": "6836ebb329459fa2e679ba808ca586ea",
```
[the remark this answers is not in the document]
Comment (nedrichards) [reply com.unity.UnityEditor.json]: Ah! I'll put it back, that's fair.

**Comment by nedrichards** [comment]
For future reference, the URL pattern for the optional components is:

```Finished downloading 'https://beta.unity3d.com/download/7b8938f7fa83/MacEditorTargetInstaller/UnitySetup-Android-Support-for-Editor-2018.2.0b2.pkg' to '/sysroot/home/nick/Downloads/UnitySetup-Android-Support-for-Editor-2018.2.0b2.pkg'
Finished downloading 'https://beta.unity3d.com/download/7b8938f7fa83/MacEditorTargetInstaller/UnitySetup-Mac-Mono-Support-for-Editor-2018.2.0b2.pkg' to '/sysroot/home/nick/Downloads/UnitySetup-Mac-Mono-Support-for-Editor-2018.2.0b2.pkg'
Finished downloading 'https://beta.unity3d.com/download/7b8938f7fa83/LinuxEditorTargetInstaller/UnitySetup-WebGL-Support-for-Editor-2018.2.0b2.tar.xz' to '/sysroot/home/nick/Downloads/UnitySetup-WebGL-Support-for-Editor-2018.2.0b2.tar.xz'
Finished downloading 'https://beta.unity3d.com/download/7b8938f7fa83/MacEditorTargetInstaller/UnitySetup-Windows-Mono-Support-for-Editor-2018.2.0b2.pkg' to '/sysroot/home/nick/Downloads/UnitySetup-Windows-Mono-Support-for-Editor-2018.2.0b2.pkg'```

**Comment by jurf** [comment]
If you make a sample extension I can populate it, I just finished school so I should have enough time.

**Comment by jurf** [comment]
Or maybe we can merge this, if it’s working alright, and add the extensions later?

**Comment by nedrichards** [comment]
The extensions are one thing - the 'how to I edit source code' is the biggest blocker.

**Comment by LeandroStanger** [comment]
How is the progress of Unity Editor Flatpak going?

**Comment by jurf** [comment]
I don’t have the time/disk space to work on it right now, maybe in a few weeks.

**Comment by nedrichards** [comment]
testbot, build com.unity.UnityEditor

**Comment by nedrichards** [comment]
Hrm, that's a good question. I know that @ramcq was interested in this so he will have a view.

**Comment by ramcq** [comment]
Hmm, it should be relatively easy to extract the AppImage into a Flatpak as an extra_data Flatpak, wouldn't it? It would provide a consistent runtime environment.

**Comment by kirbyfan64** [comment]
Okay, so good news: Unity Editor seems to still be available separately, and it can be downloaded and run!

...sort, of.

See, the bad news is that it communicates with the Unity Hub via an IPC socket (`/tmp/Unity-hubIPCService.sock`) *in order to verify the license*. Some cursory checks show that it would be pretty easy to stub this out, but at that point I'd become more concerned about TOS issues.

**Comment by jurf** [comment]
@ramcq: you mean run the Appimage in a Flatpak environment? Hm, I guess.

**Comment by nedrichards** [comment]
Superseded by #839

**Pull request closed without merging**